include/tilck/kernel/arch/riscv/sbi.h

```diff
@@ -0,0 +1,110 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#pragma once
+
+#include <tilck/common/basic_defs.h>
+
+/* Standard SBI Errors */
+#define SBI_SUCCESS                 0
+#define SBI_ERR_FAILURE            -1
+#define SBI_ERR_NOT_SUPPORTED      -2
+#define SBI_ERR_INVALID_PARAM      -3
+#define SBI_ERR_DENIED             -4
+#define SBI_ERR_INVALID_ADDRESS    -5
+#define SBI_ERR_ALREADY_AVAILABLE  -6
+#define SBI_ERR_ALREADY_STARTED    -7
+#define SBI_ERR_ALREADY_STOPPED    -8
+#define SBI_ERR_NO_SHMEM           -9
+#define SBI_ERR_INVALID_STATE      -10
+#define SBI_ERR_BAD_RANGE          -11
+
+/* Legacy Extensions */
+#define SBI_SET_TIMER                0
+#define SBI_CONSOLE_PUTCHAR          1
+#define SBI_CONSOLE_GETCHAR          2
+#define SBI_CLEAR_IPI                3
+#define SBI_SEND_IPI                 4
+#define SBI_REMOTE_FENCE_I           5
+#define SBI_REMOTE_SFENCE_VMA        6
+#define SBI_REMOTE_SFENCE_VMA_ASID   7
+#define SBI_SHUTDOWN                 8
+
+/* SBI Base Extension */
+#define SBI_BASE_EXT_ID             0x10
+#define SBI_BASE_GET_SPEC_VERSION   0
+#define SBI_BASE_GET_IMPL_ID        1
+#define SBI_BASE_GET_IMPL_VERSION   2
+#define SBI_BASE_PROBE_EXTENSION    3
+#define SBI_BASE_GET_MVENDORID      4
+#define SBI_BASE_GET_MARCHID        5
+#define SBI_BASE_GET_MIMPID         6
+
+/* System Reset (SRST) Extension */
+#define SBI_SRST_EXT_ID                  0x53525354
+#define SBI_SRST_SYSTEM_RESET            0
+#define SBI_SRST_TYPE_SHUTDOWN           0
+#define SBI_SRST_TYPE_COLD_REBOOT        1
+#define SBI_SRST_TYPE_WARM_REBOOT        2
+#define SBI_SRST_REASON_NONE             0
+#define SBI_SRST_REASON_SYSTEM_FAILURE   1
+
+#define SBI_CALL0(e, f)                     sbi_call(e, f, 0, 0, 0, 0, 0)
+#define SBI_CALL1(e, f, a0)                 sbi_call(e, f, a0, 0, 0, 0, 0)
+#define SBI_CALL2(e, f, a0, a1)             sbi_call(e, f, a0, a1, 0, 0, 0)
+#define SBI_CALL3(e, f, a0, a1, a2)         sbi_call(e, f, a0, a1, a2, 0, 0)
+#define SBI_CALL4(e, f, a0, a1, a2, a3)     sbi_call(e, f, a0, a1, a2, a3, 0)
+#define SBI_CALL5(e, f, a0, a1, a2, a3, a4) \
```

Comment:
Style: in this case the indentation for the new line is just a single one, 3 spaces:
```
#define SBI_CALL5(e, f, a0, a1, a2, a3, a4) \
   sbi_call(e, f, a0, a1, a2, a3, a4)
```

Reply:
fixed!